### DISCUSSION TITLE
How to debounce persistant?

### DISCUSSION THREAD

**Discussion by amirhhashemi** [pmndrs/zustand#1109] [Q&A] [answered]
I'm creating a rich text editor and I store the editor state in zustand. Then I use the `persist` middleware to sync the state to my backend. Something like this:
```typescript
export const useEditorStore = create()(
  persist(
    (set) => ({
      contentHtml: "",
      setContentHtml: (contentHtml) => set({ contentHtml }),
    }),
    {
      name: "editor",
      getStorage: () => AsyncStorage,
    }
  )
);
```
But I don't want to save the editor state to my DB for every state change. I want to only debounce persistent (not the whole state) with something like lodash. So how can I do that with zustand? It's probably not zustand responsibility to do such things and I have to change my strategy, but I want to know if there is an elegant way to do this with zustand. Is there a `debounce` middleware?

**Comment by dai-shi** [comment, marked as answer]
Hm, I don't think there's such a capability. You would need to provide a storge with debounce feature.
```js
      getStorage: () => debounceWrapper(AsyncStorage),
```

**Reply by amirhhashemi** [reply]
Yeah, that makes sense. Thank you.
